graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java

```diff
@@ -79,17 +89,34 @@ public ValidationResponseDTO validateQuery(
     ) {
         ValidationRequest request = prepareRequest(validationRequest, searchUser);
         final ValidationResponse response = queryValidationService.validate(request);
-        Set<ExplainResults.IndexRangeResult> searchedIndexRanges = indexRanges(request);
+        final Set<ExplainResults.IndexRangeResult> searchedIndexRanges = indexRanges(request);
+        final Optional<Set<String>> requestedStreams = validationRequest.streams();
+        final Set<String> readableStreams = request.streams();
+        final Set<DataRoutedStream> dataRoutedStreams = checkForDataRoutedStreams(requestedStreams, readableStreams, request.timerange(), request.isEmptyQuery());
+        final boolean hasDataRoutedStreams = dataRoutedStreams != null && !dataRoutedStreams.isEmpty();
+        final Optional<TimeRange> requestedRangeAsAbsolut = validationRequest.timerange().map(timeRange -> AbsoluteRange.create(timeRange.getFrom(), timeRange.getTo()));
         return ValidationResponseDTO.create(
-                toStatus(response.status(), containsWarmIndices(searchedIndexRanges)),
+                toStatus(response.status(), containsWarmIndices(searchedIndexRanges), hasDataRoutedStreams),
                 toExplanations(response),
-                searchedIndexRanges);
+                searchedIndexRanges,
+                dataRoutedStreams,
+                requestedRangeAsAbsolut);
     }
 
     private boolean containsWarmIndices(Set<ExplainResults.IndexRangeResult> searchedIndexRanges) {
         return searchedIndexRanges.stream().anyMatch(ExplainResults.IndexRangeResult::isWarmTiered);
     }
 
+    Set<DataRoutedStream> checkForDataRoutedStreams(Optional<Set<String>> requestedStreams, Set<String> readableStreams, TimeRange timeRange, boolean isEmptyQuery) {
+        if (optionalStreamQueryExplainer.isPresent()) {
```

Comment:
Minor nitpick, we could also do it like this:
```suggestion
        return optionalStreamQueryExplainer.ifPresent(explainer -> { ... }).orElse(Set.of());
```

Reply:
> Minor nitpick, we could also do it like this:

right I can use optional.map to do that

Reply:
`Optional#map` right, corrected that. :+1: